### PR TITLE
Fixes references to RFCs

### PR DESCRIPTION
**Pull request by lfrancke**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

The docs reference "DNS 952" which does not seem correct. Correct would be RFC 1035, I assume? At least I can't find anything that'd match "952".
Another one references "DNS 1123" which should be RFC 1123.

Had me confused for a while.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go [hunk 1]
```diff
@@ -41,15 +41,15 @@ func NameIsDNSSubdomain(name string, prefix bool) []string {
...
+// NameIsDNSLabel is a ValidateNameFunc for names that must be a RFC 1123 label.
 func NameIsDNSLabel(name string, prefix bool) []string {
 	if prefix {
 		name = maskTrailingDash(name)
 	}
 	return validation.IsDNS1123Label(name)
 }
 
-// NameIsDNS1035Label is a ValidateNameFunc for names that must be a DNS 952 label.
+// NameIsDNS1035Label is a ValidateNameFunc for names that must be a RFC 1035 label.
```
Comment (oomichi) [inline staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go:52]: The function IsDNS1035Label is like
```
// IsDNS1035Label tests for a string that conforms to the definition of a label in
// DNS (RFC 1035).
func IsDNS1035Label(value string) []string {
        var errs []string
        if len(value) > DNS1035LabelMaxLength {
                errs = append(errs, MaxLenError(DNS1035LabelMaxLength))
        }
        if !dns1035LabelRegexp.MatchString(value) {
                errs = append(errs, RegexError(dns1035LabelErrMsg, dns1035LabelFmt, "my-name", "abc-123"))
        }
        return errs
}
```
Then it seems fine to change 952 to 1035 at least.
`DNS 1035 label` is not so bad from the above docstring.
`label in DNS (RFC 1035)` would be another option if we want to change here.

**Review by oomichi — COMMENTED** [review]
/cc @oomichi

File: staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go (hunk 1 above)
Comment (lfrancke) [reply staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go]: I've changed the text, thanks for the review

**Review by oomichi — COMMENTED** [review]
Thanks for updating.

/ok-to-test
/lgtm

**Comment by lfrancke** [comment]
/retest

These must be unrelated (without looking) as my change is a documentation only change.

**Comment by lfrancke** [comment]
/retest

(Not sure what else to do other than retesting...)

**Pull request closed without merging**